SPM Example/PortalSwift/ViewController.swift

```diff
@@ -479,475 +626,689 @@ class ViewController: UIViewController, UITextFieldDelegate {
-      guard result.error == nil else {
-        print("❌ Error testing provider transaction request:", method, result.error ?? "")
-        completion(false)
-        return
-      }
+  @IBAction func handlePasswordRecover(_: UIButton!) {
+    Task {
+      do {
+        guard let portal = self.portal else {
+          throw PortalExampleAppError.portalNotInitialized()
+        }
+        guard let enteredPassword = await requestPassword(), !enteredPassword.isEmpty else {
+          self.logger.error("ViewController.handlePasswordRecover() - ❌ No password set by user. Recovery will not take place.")
+          return
+        }
+        guard let user else {
+          throw PortalExampleAppError.userNotLoggedIn()
+        }
+        self.startLoading()
+        // Set the Password for backup
+        try portal.setPassword(enteredPassword)
+        self.logger.debug("ViewController.handlPasswordRecover() - Starting recover...")
+        let (ethereum, solana) = try await recover(String(user.exchangeUserId), withBackupMethod: .Password)
+        guard let address = ethereum else {
+          self.logger.error("ViewController.handlePasswordRecover() - ❌ Wallet was recovered, but no address was found.")
+          throw PortalKeychain.KeychainError.noAddressesFound
+        }
+        let debugMessage = "ViewController.handlePasskeyRecover() - ✅ Wallet successfully recovered! Address: \(String(describing: address))"
+        self.logger.log(level: .debug, "\(debugMessage, privacy: .public)")
 
-      if !skipLoggingResult {
-        print("✅ ", method, "() result:", result.data?.result ?? "")
-      } else {
-        print("✅ ", method, "()")
+        DispatchQueue.main.async {
+          if let addressInformation = self.addressInformation {
+            addressInformation.text = ethereum
+          }
+        }
+        self.updateUIComponents()
+      } catch {
+        self.stopLoading()
+        self.logger.error("ViewController.handlePasskeyBackup() - Error running recover: \(error.localizedDescription)")
       }
-
-      completion(true)
     }
   }
 
-  func testProviderAddressRequest(method: String, params: [ETHAddressParam], skipLoggingResult: Bool = false, completion: @escaping (Bool) -> Void) {
-    let payload = ETHAddressPayload(
-      method: method,
-      params: params
-    )
+  // Portal API actions
 
-    print("Starting to test method ", method, "...")
-    self.portal?.provider.request(payload: payload) { (result: Result<AddressCompletionResult>) in
-      guard result.error == nil else {
-        print("❌ Error testing provider request:", method, result.error ?? "")
+  @IBAction func testGetNFTsTrxsBalancesSharesAndSimTrx() {
+    Task {
+      let chainId = "eip155:11155111"
+
+      do {
+        _ = try await self.getBalances()
+        self.logger.info("ViewController.testGetNFTsTrxsBalancesSharesAndSimTrx() - ✅ Successfully fetched balances.")
```

Comment:
Can we log the responses for these api calls to ensure it's what we're expecting?